src/channel.cr

```diff
@@ -373,8 +426,11 @@ class Channel(T)
       return ops.size, NotReady.new
     end
 
-    state = Atomic(SelectState).new(SelectState::Active)
-    contexts = ops.map &.create_context_and_wait(pointerof(state))
+    # Because channel#close may clean up a long list, `select_context.try_trigger` may
+    # be called after the select return. In order to prevent invalid address access,
+    # the state is allocated in the heap.
+    state_ptr = Pointer(Atomic(SelectState)).malloc(1, Atomic(SelectState).new(SelectState::Active))
```

Comment:
Can probably just do `Pointer.malloc` here - the compiler will guess the type from the `Atomic(SelectState).new(SelectState::Active)`.